instrumentation/src/test/java/io/opentelemetry/android/emitter/EventEmitterWithDataTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.emitter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+
+public class EventEmitterWithDataTest {
+
+    @Test
+    public void testConvertRichDataObjectIntoAttributes() {
+        Exception crashData = null;
+        try {
+            throw new RuntimeException("Crash!");
+        } catch (Exception e) {
+            crashData = e;
+        }
+
+        StringWriter stacktrace = new StringWriter();
+        crashData.printStackTrace(new PrintWriter(stacktrace));
+
+        Map<String, Object> data = new HashMap<>();
+        data.put("stringKey", "key");
+        data.put("longKey", 123L);
+        data.put("doubleKey", 12.3);
+        data.put("booleanKey", true);
+
+        data.put("stacktrace", stacktrace.toString());
+        data.put("thread", Thread.currentThread().toString());
+        data.put("throwableMessage", crashData.getMessage());
+
+        EventEmitterWithData emitter = new EventEmitterWithData(null);
+        Attributes attributes = emitter.convertRichDataObjectIntoAttributes(data);
+
+        assertEquals(attributes.size(), data.size());
+        assertEquals(data.get("stacktrace"), stacktrace.toString());
```

Comment:
Should this be ->  
```suggestion
        assertEquals(data.get("stacktrace"), attributes.get(AttributeKey.stringKey("stacktrace"));
```
and similarly for other 2 lines.

Reply:
Just updated. In practice its the same thing but this looks better for consistency sake youre right

Reply:
Just wanted to clarify for your understanding that it was not the same in practice as you were not checking what has been set in "attributes" variable instead earlier you were comparing "data" with the local variable "stacktrace" which you had set in the starting that fed data to "data"